lib/index.js

```diff
@@ -0,0 +1,237 @@
+                value: encodeURIComponent(extAttr.value),
+              });
+            }
+
+            jsonObj.extattr.push(extAttr);
+          } catch (e) {
+            debug('Failed to parse extension-attribute: %s', vCardEntry);
+          }
+          return;
+        }
+
+        if (complexJSONAttributes[JSONAttrName]) {
+          // this vCardAttribute is a complex field
+          jsonObj[JSONAttrName] = {};
+          const valArray = vCardRex[2].split(';');
+          _.forOwn(complexJSONAttributes[JSONAttrName], (val, i) => {
+            _.merge(jsonObj, {
+              [JSONAttrName]: {
+                [val]: encode
+                  ? encodeURIComponent(valArray[i])
+                  : valArray[i],
+              },
+            });
+          });
+          return;
+        }
+
+        // just a normal mapping
+        Object.assign(jsonObj, {
+          [JSONAttrName]: encode
+            ? encodeURIComponent(vCardRex[2])
+            : vCardRex[2],
+        });
+      }
+    }
+  });
+
+  return jsonObj;
+};
+
+/*
+ * @ description: Parses a JSON-Object with properties named as editableFields into a vCard 2.1 string
+ * @ param: {json} profile - the profile object
+ *
+ * @ return {string}: a vCard 2.1 string
+ *
+ * @ author: Benjamin Kroeger < benjamin.kroeger@de.ibm.com >
+ */
+VCardParser.prototype.toVcard = function toVcard(jsonObj, validAttributes) {
+  const self = this;
+  const { mappings: { toVcard: toVcardMapping }, complexJSONAttributes } = self;
+
+  // define the vCard beginning
+  const vCardArr = ['BEGIN:VCARD', 'VERSION:2.1'];
+  const parsedValidAttributes = Array.isArray(validAttributes)
+    ? validAttributes
+    : _.keys(toVcardMapping);
+
+  _.forOwn(jsonObj, (JSONAttrValue, JSONAttrName) => {
+    // take only those profile fields that are configured to be editable (information coming from users service document)
+    if (parsedValidAttributes.indexOf(JSONAttrName) > -1 && _.isString(toVcardMapping[JSONAttrName])) {
+      const vCardAttrVal = (() => {
+        if (Array.isArray(complexJSONAttributes[JSONAttrName])) {
+          const values = complexJSONAttributes[JSONAttrName].map(attrName => JSONAttrValue[attrName] || '');
+          return values.join(';');
+        }
+        if (JSONAttrName === 'extattr' && Array.isArray(JSONAttrValue)) {
+          const vCardAttrValue = [];
+
+          JSONAttrValue.forEach((extAttr) => {
+            vCardAttrValue.push(util.format(
+              'X_EXTENSION_PROPERTY;VALUE=X_EXTENSION_PROPERTY_ID:%s;VALUE=X_EXTENSION_KEY:%s;VALUE=X_EXTENSION_VALUE:%s;VALUE=X_EXTENSION_DATA_TYPE:%s', // eslint-disable-line max-len
+              extAttr.id,
+              extAttr.key,
+              _.escape(decodeURIComponent(extAttr.value)),
+              extAttr.dataType
+            ));
+          });
+          // since "vCardAttrVal" should return a string, we need to join the generated array.
+          return vCardAttrValue.join('\n');
```

Comment:
previously, the whole array was returned. What we need is a String value, separated by "\n", in order to return valid VCard document.